lib/spack/spack/solver/asp.py

```diff
@@ -1411,8 +1476,7 @@ def condition(
         imposed_spec: Optional[spack.spec.Spec] = None,
         name: Optional[str] = None,
         msg: Optional[str] = None,
-        transform_required: Optional[TransformFunction] = None,
-        transform_imposed: Optional[TransformFunction] = remove_node,
```

Comment:
I still need to double check that we maintained the same default everywhere.

Reply:
I was not doing that (although I'll note that no tests failed because of that): the latest updates restore this and also tried to make that clear (I now instantiate `ConditionContext` and set `.transform_imposed` on it in this function if it is `None`).

Reply:
> I was not doing that (although I'll note that no tests failed because of that):

That could be: `transform_imposed` is not used if `imposed_spec` is `None`.